### PR TITLE
feat: refactor base E2E tests

### PR DESCRIPTION
**Pull request by sainoe**
## Description

Closes: #2622

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API or client breaking change
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [x] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: e2e.Dockerfile [hunk 1]
```diff
@@ -11,7 +11,9 @@ RUN apk add --no-cache $PACKAGES
 RUN CGO_ENABLED=0 make install
 
 # Add to a distroless container
-FROM cgr.dev/chainguard/static:$IMG_TAG
+FROM alpine:latest
+RUN adduser -D nonroot
+RUN apk add curl
```
Comment (MSalopek) [inline e2e.Dockerfile]: Do any scripts actually use curl inside docker containers?

Comment (sainoe) [reply e2e.Dockerfile]: They don't. Good catch.

**Review by yaruwangway — APPROVED** [review]

**Review by yaruwangway — APPROVED** [review]

**Comment by yaruwangway** [comment]
LGTM!

**Review by MSalopek — APPROVED** [review]
Please run `make lint-fix` and push the changes so the linter stops complaining

**Pull request merged**